### DISCUSSION TITLE
PayPal USD discussion thread

### DISCUSSION THREAD

**Discussion by ivanmetrikin** [Cyfrin/foundry-full-course-cu#576] [General]
PayPal have just launched a new stablecoin: PYUSD

A couple of things I've observed:

- The code is written in Solidity version 0.4.24 (ancient). Nothing inherently wrong with it, just interesting it's so old 
- Seems like the code is a fork of Paxos USDP
- They can freeze and wipe user funds 
- The contract is upgradable 
- Validation checks are performed using 'require' instead of 'revert'. Not gas efficient

Ethereum contract address 0x6c3ea9036406852006290770BEdFcAbA0e23A0e8

Anything else?

**Comment by theirrationalone** [comment]
They might want more control i.e., conditional balance check, lien users accounts: freezing, upgrade with demand... It seems like they're in Hurry.

What's their interest percentage??
what's their services charges??
Why 'require' instead of 'revert'?? (hard work over smart work)

**Reply by shikhar229169** [reply]
The best thing is when I go to their website for the coin, they said its not available in my neighbourhood.
lol

**Reply by Naman1729** [reply]
Lol

**Reply by ivanmetrikin** [reply]
This is, unfortunately, not surprising from a centralised stablecoin issuer. USDC, USDT, BUSD are the same in this regard - they need a Cease and Desist function in order to operate in fiat land

**Comment by ivanmetrikin** [comment]
Code of this new stablecoin is here: https://github.com/paxosglobal/pyusd-contract#readme

**Comment by itznishant** [comment]
> PayPal have just launched a new stablecoin: PYUSD
> 
> A couple of things I've observed:
> 
> * The code is written in Solidity version 0.4.24 (ancient). Nothing inherently wrong with it, just interesting it's so old
> * Seems like the code is a fork of Paxos USDP
> * They can freeze and wipe user funds
> * The contract is upgradable
> * Validation checks are performed using 'require' instead of 'revert'. Not gas efficient
> 
> Ethereum contract address 0x6c3ea9036406852006290770BEdFcAbA0e23A0e8
> 
> Anything else?

Isn't it interesting to see the timing close to the Worldcoin launch which also wants to have its own WLD cryptocurrency token. Thoughts?
If even mainstream payment organisations are launching cryptocurrency tokens, it's obvious they are bullish on blockchains which means good times ahead for us in web3!
